src/lucky_record/form.cr

```diff
@@ -167,6 +167,18 @@ abstract class LuckyRecord::Form(T)
     end
   end
 
+  def save! : T
+    if save
+      record.not_nil!
+    else
+      raise LuckyRecord::InvalidFormError.new("#{record} has invalid parameters")
```

Comment:
I'm not quite sure how you are thinking I pass and store the form name/object. Did you mean something like this:

```crystal
class InvalidFormError(T) < Exception
  getter form_name : String
  getter form_object : T

  def initialize(@form_name : String, @form_object : T)
    super(message: "Error in #{form_name}")
  end
end
```

My Crystal's not great yet. Hopefully you understand what I mean.

Reply:
It would look something like this: https://github.com/luckyframework/record/commit/576cd3a440b6bb5095594e607198f5c0d717433c#diff-0c6732b55ddd3e3a8985f88608914664R32

And yeah I'd add getting for the form_name and form_object. You don't need to specify the type in `getter` if it is specific in the initializer. LMK if you have any other questions though. Happy to help :)